contracts/exchange/test/core.ts

```diff
@@ -1040,6 +1086,435 @@ describe('Exchange core', () => {
+                nftNotOwnerBalance,
+                // takerAddress / makerToken[0]
+                constants.INITIAL_ERC1155_FUNGIBLE_BALANCE,
+                // takerAddress / makerToken[1]
+                constants.INITIAL_ERC1155_FUNGIBLE_BALANCE,
+                // takerAddress / makerToken[2]
+                nftNotOwnerBalance,
+                // takerAddress / takerToken[0]
+                constants.INITIAL_ERC1155_FUNGIBLE_BALANCE,
+                // takerAddress / takerToken[1]
+                nftOwnerBalance,
+                // takerAddress / takerToken[2]
+                nftOwnerBalance,
+            ];
+            await erc1155Wrapper.assertBalancesAsync(tokenHolders, tokensToTransfer, expectedInitialBalances);
+            // execute transfer
+            await exchangeWrapper.fillOrderAsync(signedOrder, takerAddress, {
+                takerAssetFillAmount,
+            });
+            // check balances after transfer
+            const expectedFinalBalances = [
+                // makerAddress / makerToken[0]
+                expectedInitialBalances[0].minus(totalMakerValuesTransferred[0]),
+                // makerAddress / makerToken[1]
+                expectedInitialBalances[1].minus(totalMakerValuesTransferred[1]),
+                // makerAddress / makerToken[2]
+                expectedInitialBalances[2].minus(totalMakerValuesTransferred[2]),
+                // makerAddress / takerToken[0]
+                expectedInitialBalances[3].plus(totalTakerValuesTransferred[0]),
+                // makerAddress / takerToken[1]
+                expectedInitialBalances[4].plus(totalTakerValuesTransferred[1]),
+                // makerAddress / takerToken[2]
+                expectedInitialBalances[5].plus(totalTakerValuesTransferred[2]),
+                // takerAddress / makerToken[0]
+                expectedInitialBalances[6].plus(totalMakerValuesTransferred[0]),
+                // takerAddress / makerToken[1]
+                expectedInitialBalances[7].plus(totalMakerValuesTransferred[1]),
+                // takerAddress / makerToken[2]
+                expectedInitialBalances[8].plus(totalMakerValuesTransferred[2]),
+                // takerAddress / takerToken[0]
+                expectedInitialBalances[9].minus(totalTakerValuesTransferred[0]),
+                // takerAddress / takerToken[1]
+                expectedInitialBalances[10].minus(totalTakerValuesTransferred[1]),
+                // takerAddress / takerToken[2]
+                expectedInitialBalances[11].minus(totalTakerValuesTransferred[2]),
+            ];
+            await erc1155Wrapper.assertBalancesAsync(tokenHolders, tokensToTransfer, expectedFinalBalances);
+        });
+        it('should allow an order exchanging erc1155 assets to be partially filled', async () => {
+            // NOTICE:
+            // As-per the eip1155 standard, there is no way to distinguish between a fungible or non-fungible erc1155 assets.
```

Comment:
Shouldn't a partial transfer of a non-fungible still fail with `TRANSFER_FAILED`? Whatever the behavior is, we should add a test for it.

Reply:
We chatted about this offline.

Partial fills on non-fungible assets should be prohibited by the maker by setting the amount to 1. The proxy cannot distinguish between fungible and non-fungible assets, so it cannot force partial fills to fail in this scenario. If the amount is _not_ set to 1 and an asset is non-fungible then the behavior depends on the implementation of the ERC1155 contract.